### PR TITLE
Remove duplicate GetLatestEpoch

### PR DESCRIPTION
**Pull request by gdbelvin**
Remove a duplicate, and inconsitent call to getLatestLogRoot.
Previously, GetLatestEpoch had a flow like this:

 * GetLatestLogRoot
 * Calculate LatestMapRevision
 * **GetLatestLogRoot** (which could be different!)
 * GetMapRevision

### CODE REVIEW

File: core/keyserver/epochs.go [hunk 1]
```diff
@@ -192,11 +184,8 @@ func (s *Server) logProofs(ctx context.Context, d *domain.Domain, firstTreeSize
 		glog.Errorf("logProofs(): log.GetInclusionProof(%v, %v, %v): %v", d.LogID, epoch, secondTreeSize, err)
```
Comment (AlCutter) [inline core/keyserver/epochs.go]: Do you need to update the error message to reflect the new func name?

File: core/client/get_and_verify.go [hunk 2]
```diff
@@ -77,7 +77,7 @@ func (c *Client) VerifiedGetLatestEpoch(ctx context.Context) (*types.LogRootV1,
 		return nil, nil, err
 	}
 	if smr.Revision != wantRevision {
-		return nil, nil, fmt.Errorf("GetLatestEpoch() did not return latest map revision. Got MapRoot.Revison: %v, want: %v", smr.Revision, wantRevision)
+		return nil, nil, fmt.Errorf("GetLatestEpoch(): smr.Revison: %v + 1 != slr.TreeSize: %v", smr.Revision, slr.TreeSize)
```
Comment (AlCutter) [inline core/client/get_and_verify.go]: Would the message `("... smr.Revision (%v) != slr.TreeSize-1 (%v)", smr.Revision, wantRevision)` better represent what's actually being tested?

File: core/keyserver/epochs.go (hunk 1 above)
Comment (gdbelvin) [reply core/keyserver/epochs.go]: done

File: core/client/get_and_verify.go (hunk 2 above)
Comment (gdbelvin) [reply core/client/get_and_verify.go]: yep

**Comment by gdbelvin** [comment]
PTAL

**Pull request merged**